pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -722,32 +769,35 @@ int kprobe__tcp_close(struct pt_regs* ctx) {
 
     log_debug("kprobe/tcp_close: pid_tgid: %d, ns: %d\n", pid_tgid, net_ns_inum);
 
-    if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_TCP)) {
+    if (!read_conn_tuple(&t, NULL, NULL, sk, pid_tgid, CONN_TYPE_TCP)) {
         return 0;
     }
 
-    cleanup_tcp_conn(ctx, &t);
+    cleanup_conn(&t);
     return 0;
 }
 
-SEC("kretprobe/tcp_close")
-int kretprobe__tcp_close(struct pt_regs* ctx) {
+static __always_inline void flush_conn_close_batch(struct pt_regs * ctx) {
```

Comment:
Let's call this flush_conn_close_if_full since this doesn't always actually do a flush